### PR TITLE
(ManagerV2): issue on firmware update reset connnection

### PR DESCRIPTION
**Pull request by LFBarreto**
While on firmware update on the bootloader phase that asks you to disconnect and reconnect your device, the manager state was resetted thus looping beetween the osu download and the device connection.

localhost_8080_webpack_index html

The device connection state is now locked on firmware update and controlled by the manager dashboard component.

### Type

Bug Fix

### Context

maybe connected to LL-2178

### Parts of the app affected / Test plan

ManagerV2 mainly on firmware updates

### CODE REVIEW

File: src/renderer/screens/manager/FirmwareUpdate/index.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ import type { ModalStatus } from "./types";
 type Props = {
   deviceInfo: DeviceInfo,
   device: Device,
+  onUpdateStart: () => void,
```
Comment (valpinkman) [inline src/renderer/screens/manager/FirmwareUpdate/index.js:23]: are they always required ? Are you sure there are no other context in which onUpdateStart and onUpdateStop will not get passed ?

Comment (valpinkman) [reply src/renderer/screens/manager/FirmwareUpdate/index.js:23]: My bad I thought it was the Modal

**Review by valpinkman — APPROVED** [review]

File: src/renderer/components/ManagerConnect/index.js [hunk 2]
```diff
@@ -292,6 +298,11 @@ const ManagerConnect = ({
     );
   }
 
+  if (onSuccess) {
+    onSuccess({ device, deviceInfo, result });
+    return null;
+  }
```
Comment (gre) [inline src/renderer/components/ManagerConnect/index.js:304]: ok, i apologize that this was not existing yet. unfortunately the current implementation is going to create bugs (could double call onSuccess. not a good idea to call in render step). please look and try to get in this implementation instead: https://github.com/LedgerHQ/lld-v2/blob/poc-device-connect/src/renderer/components/DeviceAction/index.js#L195

**OR maybe we should just get this in the PR #127**

File: src/renderer/screens/manager/Dashboard.js [hunk 3]
```diff
@@ -10,15 +11,32 @@ import TrackPage from "~/renderer/analytics/TrackPage";
...
 };
 
-const Dashboard = ({ device, deviceInfo, result }: Props) => {
+const Dashboard = ({ device, deviceInfo, result, onReset }: Props) => {
+  /** locks and unlocks the device connection state reset ie: firmware update */
+  const [resetLock, setResetLock] = useState(false);
+  const lockReset = useCallback(() => setResetLock(true), [setResetLock]);
+  const unlockReset = useCallback(() => setResetLock(false), [setResetLock]);
+  /** we listen to the redux state to detect the current device connected */
+  const currentDevice = useSelector(getCurrentDevice);
```
Comment (gre) [inline src/renderer/screens/manager/Dashboard.js:29]: as this component seems to be connect, I would suggest instead to just have:

```
resetLock = useSelect(aSelectorThatCheckIfFirmwareUpdateModalIsOpened)
```

that would simplify some code below.

**Review by gre — CHANGES_REQUESTED** [review]

**Comment by gre** [comment]
@LFBarreto let's work together tomorrow to see if we can backport this work directly inside the DeviceAction rework PR (#127) because things will be simpler there. also i think my comment regarding detecting a firmware update by modal presence could work but to be double check ;)

**Comment by gre** [comment]
integrated in https://github.com/LedgerHQ/lld-v2/pull/127/commits/36ea62b95e324a7bb0f3bcf7d06f33461a1c4cb0
in context of https://github.com/LedgerHQ/lld-v2/pull/127

**Comment by gre** [comment]
there are other changes in this PR unrelated to firmware update, please resubmit them in another PR @LFBarreto

**Pull request closed without merging**